src/helpers/getLoggedInUser.js

```diff
@@ -0,0 +1,30 @@
+import axios from '@nextcloud/axios'
+import { generateUrl } from '@nextcloud/router'
+
+/**
+ * Gets the current user's display name if logged in.
+ *
+ * @return null | string
+ */
+async function getLoggedInUser() {
+	try {
```

Comment:
Just some background on why this worked before and why it now doesn't:

The files_sharing app [sets the incognito mode](https://github.com/nextcloud/server/blob/e7a7b4a40184dc3da2c83e858c820625f660e48e/apps/files_sharing/lib/Controller/ShareController.php#L327) which will basically [hide the user id](https://github.com/nextcloud/server/blob/ec5133b739eabc76271789504b4dbb91a534f552/lib/private/User/Session.php#L231) in any code that tries to fetch it, so also in the template rendering logic that would normally add the `data-user` attributes to the html attribute.

With the previous logic of document.js being nested in another iframe this frame document didn't apply the incognito mode so within the iframe we knew that the user was still logged in. 

The incognito mode was introduced in https://github.com/owncloud/core/pull/6003 however I think has quite some design flaws with hiding the user id.

I think your approach works fine, we could do it slightly different if we want to save the extra request:
- If we don't need the user id in Office.vue (the token api request will still have it) we can just expose an initial state of the incognito mode and skip the guest name picker then
```diff
diff --git a/lib/Service/InitialStateService.php b/lib/Service/InitialStateService.php
index 6726a09d..ee653ac2 100644
--- a/lib/Service/InitialStateService.php
+++ b/lib/Service/InitialStateService.php
@@ -94,6 +94,7 @@ class InitialStateService {
        }

        private function provideOptions(): void {
+               $this->initialState->provideInitialState('incognitoMode', \OC_User::isIncognitoMode());
                $this->initialState->provideInitialState('theme', $this->config->getAppValue(Application::APPNAME, 'theme', 'nextcloud'));
                $this->initialState->provideInitialState('uiDefaults', [
                        'UIMode' => $this->config->getAppValue(Application::APPNAME, 'uiDefaults-UIMode', 'notebookbar')
```
- If we need the user information we can do something like this: https://github.com/nextcloud/richdocuments/blob/5bd5e09473ccb94a524d5b581845a1558183e575/lib/PermissionManager.php#L65-L73